src/BuildingRegistry/Building/Building.cs

```diff
@@ -375,7 +375,8 @@ private void UpdateStatusCommonBuildingUnit()
                 ApplyChange(new BuildingUnitWasCorrectedFromRetiredToRealized(
                     BuildingPersistentLocalId,
                     commonBuildingUnit.BuildingUnitPersistentLocalId,
-                    commonBuildingUnitPosition));
+                    BuildingGeometry.Center,
```

Comment:
A common building unit is always DerivedFromObject, so it's position should only be updated if the building geometry was updated. Hence commonBuildingUnitPosition is null when the building geometry hasn't changed.